### PR TITLE
TrustyAI service KServe integration

### LINKED ISSUE #356: KServe CloudEvent consumer

### PR DESCRIPTION
**Pull request by ruivieira**
This PR adds the ability for the TrustyAI service to work as a KServe `InferenceLogger`, consume inference payloads (inputs and ouptuts), reconcile them and save them to storage. This allows TrustyAI to use KServe `InferenceServices` for bias/fairness metrics similarly to ModelMesh `InferenceServices`.

To use TrustyAI as inference logger the section

```
spec:
  predictor:
    logger:
      mode: all
      url: http://trustyai-service
```

must be added to the `InferenceService` manifest. [In a separate PR](https://github.com/trustyai-explainability/trustyai-explainability/issues/362), this will be handled by the TrustyAI operator.

## Changes details

- Add a consumer and parser for KServe `CloudEvents`
- Since the ModelMesh and KServe reconciler share much of the code, an abstract `PayloadReconciler` is added, with specific implementations for ModelMesh and KServe
- The `PartialPayload` interface is added, along with implementations for partial KServe and ModelMesh payload, so that the generic payload reconciler can make use of it

### CODE REVIEW

File: explainability-service/pom.xml [hunk 1]
```diff
@@ -170,6 +176,19 @@
...
+                <version>3.1.2</version>
+                <executions>
+                    <execution>
+                        <id>make-index</id>
+                        <goals>
+                            <goal>jandex</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
```
Comment (tteofili) [inline explainability-service/pom.xml]: is this plugin needed for knative/funqy integration?

**Review by tteofili — APPROVED** [review]
LGTM with just a minor comment

File: explainability-service/pom.xml (hunk 1 above)
Comment (ruivieira) [reply explainability-service/pom.xml]: Removed

File: explainability-service/src/main/java/org/kie/trustyai/service/data/utils/ModelMeshInferencePayloadReconciler.java [hunk 2]
```diff
@@ -121,8 +121,8 @@ public Dataframe payloadToDataFrame(byte[] inputs, byte[] outputs, String id, Ma
         LOG.debug("Prediction input: " + predictionInput.getFeatures());
         final List<Feature> features = new ArrayList<>(predictionInput.getFeatures());
 
-        DatapointSource datapointSource = metadata.containsKey(ExplainerEndpoint.BIAS_IGNORE_PARAM) ? DatapointSource.SYNTHETIC : DatapointSource.UNLABELED;
-        PredictionMetadata predictionMetadata = new PredictionMetadata(id, LocalDateTime.now(), datapointSource);
+        String datapointTag = metadata.containsKey(ExplainerEndpoint.BIAS_IGNORE_PARAM) ? Dataframe.InternalTags.SYNTHETIC.get() : "";
```
Comment (RobGeada) [inline explainability-service/src/main/java/org/kie/trustyai/service/data/utils/ModelMeshInferencePayloadReconciler.java]: should be `Dataframe.InternalTags.SYNTHETIC.get() : Dataframe.InternalTags.UNLABELED.get()`

Comment (ruivieira) [reply explainability-service/src/main/java/org/kie/trustyai/service/data/utils/ModelMeshInferencePayloadReconciler.java]: Done

**Pull request merged**